File: 백준/Silver/9012. 괄호/괄호.java
import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.IOException;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;
//import java.util.Stack;

public class Main {
	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
		
		int N = Integer.parseInt(br.readLine());
		
		//Stack<Integer> stack = new Stack<>();
		
		for (int i = 0; i < N; i++) {
			String str = br.readLine();
			
			while (true) {
				if (str.contains("()")) str = str.replace("()", "");
				else break;
			}
			
			if (str.equals("")) System.out.println("YES");
			else System.out.println("NO");
		}
	}
}

File: 백준/Silver/9012. 괄호/괄호.py
import sys

def yes_or_no(string):
    while "()" in string:
        string = string.replace("()", "")
    if string == "":
        print("YES")
    else:
        print("NO")
        
if __name__ == "__main__":
    N = int(sys.stdin.readline())

    array = []

    for i in range(N):
        order = sys.stdin.readline().strip()
        array.append(order)

    for i in array:
        yes_or_no(i)